### ISSUE TITLE
Allow specification for file extension to be used in import statement paths for files created by code generation

### ISSUE THREAD

**Issue by TheMartianMaker** [reduxjs/redux-toolkit#4645]
Feature Request:
- Option to specify a file extension: Add a configuration option to specify the file extension (e.g., .js, .ts) that should be used in all import statements within the generated files.
- Consistent import paths: Ensure that all import statements in the generated files use the specified file extension.

Use Case:
For ESM projects, it is often required to include file extensions in import statements. Currently, developers have to manually update the import statements in the generated files to include the necessary file extensions. This feature would automate that process and ensure consistency across the codebase, which is crucial for maintaining proper module resolution in ESM environments.

**Comment by markerikson** [comment]
Seems reasonable. File a PR?

**Comment by TheMartianMaker** [comment]
I dont' have the code written for this request.  I inquiring to see if you guys could add this feature.

**Comment by markerikson** [comment]
@TheMartianMaker Yeah, I'm saying it's a reasonable request, but it's probably not something we're going to have time to implement ourselves any time soon. Best bet to add it soon is if you can file a PR to make the changes, otherwise it'll probably be a while.

**Comment by markerikson** [comment]
@TheMartianMaker no, the codegen has been low priority - all my own time atm is focused on pushing the infinite query feature forward.

**Comment by markerikson** [comment]
Loosely, yeah.

I've learned over the years that while I can absorb a lot of info, when it comes to large projects I can only deal with one at a time.  That's especially true with something like a large RTK feature, where I only find time to work on it intermittently.  It's also a matter of prioritization.

**Comment by markerikson** [comment]
Out in https://github.com/reduxjs/redux-toolkit/releases/tag/%40rtk-query%2Fcodegen-openapi%402.1.0 !
